### PR TITLE
Af xdp bug 6238

### PR DESCRIPTION
**Pull request by joser12345678**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [JR ] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [JR ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ JR] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6238

Describe changes:
Added a RunModeAFXDPRemoveProg function to remove an AF_XDP program before closing sockets as this prevents the crash when receiving high traffic volume and shutting down suricata.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by jufajardini** [comment]
Previous PR: https://github.com/OISF/suricata/pull/9319

**Review by jufajardini — CHANGES_REQUESTED** [review]
Hi, thanks for splitting the commits, incorporating the feedback, and submitting the changes in a new PR. 

When addressing feedback, please add a version to your PR, and mention the previous one, so it's easier the follow the changes, during the reviewing process.

We try to keep our commit history as clean as possible, so please, if you have a commit that introduces a change and a subsequent commit that removes said change in the same PR, just squash them (that is, don't introduce a change that you will just remove later ;) ).

I noticed that you adjusted the commit message. I apologize, my wording wasn't very clear. In the commit title, instead of "module", use the module name you have added changes to. Considering what I'm seeing, I think you could use `af/xdp` for module name.

In the commit message itself, please break the lines at ~72 characters.

These indications are in our contribution guidelines, but I know they aren't very clear. We're trying to improve there...

Do you think it's possible to craft a [Suricata-verify test](https://github.com/OISF/suricata-verify#readme) for this? I'm guessing it's not, but want to be sure...

File: src/runmode-af-xdp.c [hunk 1]
```diff
@@ -368,7 +408,7 @@ int RunModeIdsAFXDPWorkers(void)
     RunModeInitialize();
     TimeModeSetLive();
 
-    (void)ConfGet("af-xdp.live-interface", &live_dev);
+    ret = ConfGet("af-xdp.live-interface", &live_dev);
```
Comment (jufajardini) [inline src/runmode-af-xdp.c:411]: I'm not sure I understand why to get this return value here, if we won't use it.

**Comment by jufajardini** [comment]
> Hi, thanks for splitting the commits, incorporating the feedback, and submitting the changes in a new PR.
> 

The author name also still needs fixing in the commits, if you're unsure how to do this, I find this guide helps: https://www.git-tower.com/learn/git/faq/change-author-name-email

**Comment by joser12345678** [comment]
Ok will fix these things in the next PR.

Also it is hard to create a test to verify this as I would need to generate lots of test traffic. So I don't believe there is a good way to do that

**Pull request closed without merging**